PermanentOrg/web-app
# Add eslint ruleset with failing rules disabled

This PR installs some eslint plugin/config packages that we use in our other TypeScript repos, enables them in eslint, and then disables every single failing eslint rule. Once we get eslint enabled in CI, we can then begin the slow process of individually turning on rules and fixing the code that breaks these rules.

I've marked off the places where we want to enable rules vs permanent rules we want to keep in the `.eslintrc.js` file (that has been converted from the `.json` format so we can put comments in the file). I've also alphabetized the list of rules just so we work on them in some methodical order, although I pulled the "import" ruleset to the top of the list since those looked like good first rules to tackle (they aren't likely to break existing functionality, and if they do break it, it will cause compilation errors rather than introduce subtle bugs).

**Steps to test**
1. Run `npm install`
2. Run `npm run lint`
3. Verify that there are no linting errors.
